### PR TITLE
fix: auto-join to guild if user is not member

### PR DESCRIPTION
**Pull request by arashsheyda**
The user will be join to the server if is not a member.
related to #13

### CODE REVIEW

**Comment by arashsheyda** [comment]
and maybe we could add `guilds.join` to https://github.com/nuxt/nuxters/blob/a57a22faac2361f9c31c686147048fb6363cc511/server/routes/connect/%5Bprovider%5D.get.ts#L35 to inform user:

image

File: server/routes/oauth/discord.ts [hunk 1]
```diff
@@ -51,6 +51,25 @@ export default defineEventHandler(async event => {
...
+  // makes sure the user is in the guild
+  if (!session.data.guildMemberAdded) {
+    console.log('adding user to guild', session.data.guildMemberAdded)
+    await $fetch(
+      `https://discord.com/api/guilds/${config.discord.guildId}/members/${session.data.discordId}`,
+      {
+        method: 'PUT',
+        body: JSON.stringify({
+          access_token,
+        }),
```
Comment (manniL) [inline server/routes/oauth/discord.ts:63]: nit: no stringify needed when using `$fetch` 😋

```suggestion
        body: {
          access_token,
        },
```

**Review by manniL — APPROVED** [review]

**Comment by manniL** [comment]
I think adding `guild.join` to the scope would be good too, yes 👍🏻

File: server/routes/oauth/discord.ts (hunk 1 above)
Comment (arashsheyda) [reply server/routes/oauth/discord.ts:63]: aw yeah I forgot how great is $fetch. thanks

**Review by manniL — APPROVED** [review]

**Comment by Atinux** [comment]
Seems it's completely broken now in production

**Comment by manniL** [comment]
@Atinux Indeed, getting a 403 too. We need to extend the `scope` (to add `guilds.join`) in the Discord app I think.

**Comment by arashsheyda** [comment]
@Atinux @manniL Oh sorry. I tested in another server before pushing and it did work, 
so sorry for the inconvenience

Nuxter bot is in Nuxt server right? I think if the bot is not in server we might get 403

**Comment by danielroe** [comment]
I think we're good now 👌

**Pull request merged**